hydroflow/examples/chat/client.rs

```diff
@@ -1,112 +1,91 @@
 use chrono::prelude::*;
 use colored::Colorize;
 
-use crate::protocol::{ChatMessage, MemberRequest, MemberResponse};
+use crate::protocol::{deserialize_msg, serialize_msg, Message};
 use crate::{GraphType, Opts};
-use hydroflow::builder::prelude::*;
-use hydroflow::scheduled::handoff::VecHandoff;
+use chrono::Utc;
+use hydroflow::hydroflow_syntax;
+use std::net::SocketAddr;
+use tokio::io::AsyncBufReadExt;
+use tokio::net::UdpSocket;
+use tokio_stream::wrappers::LinesStream;
 
 pub(crate) async fn run_client(opts: Opts) {
-    let mut hf = HydroflowBuilder::default();
+    // set up network and I/O channels
+    let server_addr: SocketAddr = format!("{}:{}", opts.addr, opts.port).parse().unwrap();
 
-    // setup connection req/resp ports
-    let connect_req = hf.hydroflow.outbound_tcp_vertex::<MemberRequest>().await;
-    let connect_req = hf.wrap_output(connect_req);
-    let (connect_response_port, connect_resp) =
-        hf.hydroflow.inbound_tcp_vertex::<MemberResponse>().await;
-    let connect_resp = hf.wrap_input(connect_resp);
+    let client_socket = UdpSocket::bind("127.0.0.1:0").await.unwrap();
+    let client_addr = client_socket.local_addr().unwrap();
+    let (outbound, inbound) = hydroflow::util::udp_lines(client_socket);
 
-    // setup message send/recv ports
-    let (messages_port, messages_recv) = hf.hydroflow.inbound_tcp_vertex::<ChatMessage>().await;
-    let messages_recv = hf.wrap_input(messages_recv);
-    let messages_send = hf.hydroflow.outbound_tcp_vertex().await;
-    let messages_send = hf.wrap_output(messages_send);
+    let reader = tokio::io::BufReader::new(tokio::io::stdin());
+    let stdin_lines = LinesStream::new(reader.lines());
+    println!("Client live!");
 
-    // setup stdio input handler
-    let text_out = {
-        use futures::stream::StreamExt;
-        use tokio::io::AsyncBufReadExt;
-        let reader = tokio::io::BufReader::new(tokio::io::stdin());
-        let lines = tokio_stream::wrappers::LinesStream::new(reader.lines())
-            .map(|result| Some(result.expect("Failed to read stdin as UTF-8.")));
-        hf.add_input_from_stream::<_, _, VecHandoff<String>, _>("stdin", lines)
-    };
-
-    // format addresses
-    let addr = format!("localhost:{}", opts.port);
-    let connect_addr = format!("localhost:{}", connect_response_port);
-    let messages_addr = format!("localhost:{}", messages_port);
+    let mut hf = hydroflow_syntax! {
+        // set up channels
+        outbound_chan = merge() -> sink_async(outbound);
+        inbound_chan = recv_stream(inbound) -> map(deserialize_msg) -> tee();
+        connect_acks = inbound_chan[0] -> filter_map(|m: Message| match m {
+            Message::ConnectResponse => Some(m),
+            _ => None }) -> tee();
+        messages = inbound_chan[1] -> filter_map(|m: Message| match m {
+            Message::ChatMsg{ nickname, message, ts } =>
+                Some(Message::ChatMsg{nickname, message, ts}),
+            _ => None });
```

Comment:
Yeah can put a `//TODO` here for when we have an actual switching/demuxing operator

Reply:
I chose a different demuxing style for now, it's clear what's going on. The TODO is implicit!